### PR TITLE
Basic setup of sequelize with postgres

### PR DESCRIPTION
**Pull request by avemike**

### CODE REVIEW

File: src/services/ConfigService.ts [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 import { FastifyServerOptions } from 'fastify';
+import { SequelizeOptions } from 'sequelize-typescript';
 import ConfigValidationError from '../errors/ConfigValidationError';
 import { logger } from '../logger';
+import databaseLogger from '../logger/databaseLogger';
 
 type Config = {
   HTTP_PORT: number | string;
   LISTENING_IP: string;
+  database: Record<string, any>;
```
Comment (Dziejo93) [inline src/services/ConfigService.ts:12]: ad ts-ignore or unknown

File: src/database/index.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  constructor(options: SequelizeOptions) {
+    super(options);
+  }
+
+  async connect() {
+    await this.authenticate();
+
+    // @todo: Remove after first release
+    if (process.env.NODE_ENV === 'development') {
+      await this.sync({ force: true });
```
Comment (Dziejo93) [inline src/database/index.ts:14]: .

**Review by Dziejo93 — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -26,23 +27,31 @@
...
-    "eslint": "^7.7.0",
+    "eslint": "^7.2.0",
+    "eslint-config-airbnb": "^18.2.0",
     "eslint-config-prettier": "^6.11.0",
     "eslint-import-resolver-typescript": "^2.2.1",
-    "eslint-plugin-import": "^2.22.0",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.3.1",
     "eslint-plugin-prettier": "^3.1.4",
+    "eslint-plugin-react": "^7.21.5",
```
Comment (Dziejo93) [inline package.json:48]: : |

File: package.json [hunk 4]
```diff
@@ -26,23 +27,31 @@
...
     "@typescript-eslint/eslint-plugin": "^4.0.1",
     "@typescript-eslint/parser": "^4.0.1",
-    "eslint": "^7.7.0",
+    "eslint": "^7.2.0",
+    "eslint-config-airbnb": "^18.2.0",
     "eslint-config-prettier": "^6.11.0",
     "eslint-import-resolver-typescript": "^2.2.1",
-    "eslint-plugin-import": "^2.22.0",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.3.1",
```
Comment (Dziejo93) [inline package.json:46]: : |

File: package.json [hunk 5]
```diff
@@ -26,23 +27,31 @@
...
     "eslint-plugin-prettier": "^3.1.4",
+    "eslint-plugin-react": "^7.21.5",
+    "eslint-plugin-react-hooks": "^4.0.0",
```
Comment (Dziejo93) [inline package.json:49]: : |

File: package.json [hunk 6]
```diff
@@ -26,23 +27,31 @@
...
+    ]
   },
   "devDependencies": {
     "@types/jest": "^26.0.12",
     "@types/node": "^14.6.2",
     "@typescript-eslint/eslint-plugin": "^4.0.1",
     "@typescript-eslint/parser": "^4.0.1",
-    "eslint": "^7.7.0",
+    "eslint": "^7.2.0",
+    "eslint-config-airbnb": "^18.2.0",
```
Comment (Dziejo93) [inline package.json:42]: imo this is not good idea:
a) we have prettier
b) airbnb was iffy for ts

File: src/database/index.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  async connect() {
+    await this.authenticate();
+  }
+
+  async disconnect() {
+    return this.close();
+  }
+
+  getModel<M>(modelName: string): ModelCtor<Model<M>> {
```
Comment (Dziejo93) [inline src/database/index.ts:17]: hmm could you try to type modelName stricter ?

File: src/database/migrate.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    },
+  });
+
+const migrator = umzugFactory('migrations');
+const seeder = umzugFactory('seeders');
+
+const task = (process.argv[2] || '').trim();
+
+if (task === 'up') {
+  migrator.up().then(() => {
```
Comment (Dziejo93) [inline src/database/migrate.ts:26]: can't you use async wait ?

File: tests/integration/smoke.ts [hunk 9]
```diff
@@ -1,7 +1,32 @@
...
+    await db.sync({ force: true });
+
+    tcpProxy = TcpProxy.createProxy(8181, 'postgres-test', 5432, { quiet: true });
+
+    const user = (await db.models.User.create({
+      email: 'naturalnyjanusztestowy@gmail.com',
+      name: 'Januszek',
+    })) as User;
+    const userRaw = user.toJSON() as User;
+    userAccessor.set({ ...userRaw });
```
Comment (Dziejo93) [inline tests/integration/smoke.ts:21]: userAccessor.set(userRaw);

File: tests/utils/userAccessor.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+type UserInput = {
+  id: number;
+  email: string;
+  name: string;
+};
+
+export const userAccessor = {
+  value: null,
+  set(input: UserInput): void {
+    this.value = { ...input };
```
Comment (Dziejo93) [inline tests/utils/userAccessor.ts:10]: .

**Review by kvcpr — APPROVED** [review]

**Pull request merged**